### PR TITLE
Add a blog post for LIKE optimizations

### PR DESCRIPTION
**Pull request by xumingming**

### CODE REVIEW

**Comment by xumingming** [comment]
cc @mbasmanova

File: website/blog/2024-01-27-like-optimization.mdx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+optimizations for some other patterns:
+
+- `%hello`: matches inputs that start with `hello`. It can be optimized by memory comparing the suffix bytes of the inputs.
+- `%hello%`: matches inputs that contain `hello`. It can be optimized by using `std::string_view::find` to check whether inputs contain `hello`.
+
+These simple patterns are straightforward to optimize, there are some more
+relaxed patterns that are not so straightforward:
+
+- `hello_velox%`: matches inputs that start with 'hello', followed by any character, then followed by 'velox'.
+- `hello_velox%`: matches inputs that end with 'hello', followed by any character, then followed by 'velox'.
```
Comment (lingbin) [inline website/blog/2024-01-27-like-optimization.mdx:64]: typo: `hello_velox%` -> `%hello_velox` ?

Comment (xumingming) [reply website/blog/2024-01-27-like-optimization.mdx]: Nice catch! Fixed.

**Review by mbasmanova — APPROVED** [review]
@xumingming James, thank you for writing this nice. It reads very well. 

@pedroerp Pedro, would you also take a look?

File: website/blog/2024-01-27-like-optimization.mdx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  }
+}
+```
+
+Here `cursor` is the index in the input we are trying to match, `unicodeCharLength` is
+a function which wraps utf8proc function to determine how many bytes current character consists of,
+so the logic is basically repeatedly calculate size of current character and skip it.
+
+It seems not that complex, but we should note that this logic is not effective for pure ASCII input,
+for pure ASCII input, every character is one byte, to match a sequence of `_`, we don't need to
```
Comment (aditi-pandit) [inline website/blog/2024-01-27-like-optimization.mdx:125]: Nit sentence :
"Every character is one byte in pure ASCII input. So to match a sequence of '_', we don't need to calculate the size of each character and compare in a for-loop. Infact, we don't need to explicitly match '_' for pure ASCII input as all. We can use the following logic instead:"

File: website/blog/2024-01-27-like-optimization.mdx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        std::memcmp(
+            input.data() + start + subPattern.start,
+            patternMetadata.fixedPattern().data() + subPattern.start,
+            subPattern.length) != 0) {
+      return false;
+    }
+}
+```
+
+It only matches the kLiteralString pattern at the right position of the inputs, `_` is automatically
```
Comment (aditi-pandit) [inline website/blog/2024-01-27-like-optimization.mdx:141]: End this sentence with full-stop.

**Review by aditi-pandit — COMMENTED** [review]
Thanks @xumingming for this blog post. It is very useful.

Had some nits in the writing.

**Review by Yuhta — APPROVED** [review]

**Comment by xumingming** [comment]
@aditi-pandit Thanks for the review, made corresponding changes to all the comments.

**Pull request closed without merging**